packages/dart/lib/src/telemetry_processing/telemetry_buffer.dart

```diff
@@ -0,0 +1,42 @@
+import 'dart:async';
+
+import '../../sentry.dart';
+import 'json_encodable.dart';
+
+/// A buffer that batches telemetry items for efficient transmission to Sentry.
+///
+/// Collects items of type [T] and sends them in batches rather than
+/// individually, reducing network overhead.
+abstract class TelemetryBuffer<T> {
+  /// Adds an item to the buffer.
+  void add(T item);
+
+  /// Manually sends all buffered items to Sentry and clears the buffer.
+  FutureOr<void> flush();
```

Comment:
Didn't check in the other PR, but in Cocoa we had the discussion if this should be called `clear` (or similar) instead of flush, up to you if you want to change this.

Reply:
done